### PR TITLE
Fix example spec validation errors

### LINKED ISSUE #100: Example spec has schema validation errors
https://editor.swagger.io registers the following errors in spec.template.yaml:

Schema error at paths['/robots.txt'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/'].get
should have required property 'responses'
missingProperty: responses

Semantic error at paths./{domain}/v1/siteinfo{/prop}
Declared path parameter "domain" needs to be defined as a path parameter at either the path or operation level

Semantic error at paths./{domain}/v1/siteinfo{/prop}
Declared path parameter "/prop" needs to be defined as a path parameter at either the path or operation level

Schema error at paths['/{domain}/v1/siteinfo{/prop}'].get
should have required property 'responses'
missingProperty: responses

Semantic error at paths./{domain}/v1/page/{title}
Declared path parameter "domain" needs to be defined as a path parameter at either the path or operation level

Semantic error at paths./{domain}/v1/page/{title}
Declared path parameter "title" needs to be defined as a path parameter at either the path or operation level

Schema error at paths['/{domain}/v1/page/{title}'].get
should have required property 'responses'
missingProperty: responses

Semantic error at paths./{domain}/v1/page/{title}/lead
Declared path parameter "domain" needs to be defined as a path parameter at either the path or operation level

Semantic error at paths./{domain}/v1/page/{title}/lead
Declared path parameter "title" needs to be defined as a path parameter at either the path or operation level

Schema error at paths['/{domain}/v1/page/{title}/lead'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/_info'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/_info/name'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/_info/version'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/_info/home'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/ex/err/array'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/ex/err/file'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/ex/err/manual/error'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/ex/err/manual/deny'].get
should have required property 'responses'
missingProperty: responses

Schema error at paths['/ex/err/manual/auth'].get
should have required property 'responses'
missingProperty: responses

### PR DESCRIPTION
**Pull request by mdholloway**
Fixes validation errors identified by editor.swagger.io.

The only remaining error pertains to the optional 'prop' path parameter.
Optional path parameters are not legal according to the spec but are
handled by the swagger-ui module.

Fixes #100

### CODE REVIEW

File: spec.template.yaml [hunk 1]
```diff
@@ -68,6 +99,23 @@ paths:
       description: Calls the MW API siteinfo action and returns the response
       produces:
         - application/json
+      parameters:
```
Comment (Pchelolo) [inline spec.template.yaml:102]: I think that describing parameters like this was only added in 3.0 spec version, in 2.0 it's not supported.

Comment (mdholloway) [reply spec.template.yaml:102]: No?  https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#operationObject says that operation objects can have  a 'parameters' field that can take reference objects, and the 2.0 version of the online editor (http://editor2.swagger.io/) seems fine with it.

Comment (Pchelolo) [reply spec.template.yaml:102]: Oh ok. Didn't know that. We've been describing parameters inline mostly, good to know.

I assume you've checked how the swagger-ui looks with these changes and all good? If so LGTM

Comment (mdholloway) [reply spec.template.yaml:102]: Yes, the swagger-ui still looks good.

**Pull request merged**